extensions/wikia/AdEngine/js/context/adContext.js

```diff
@@ -42,11 +42,67 @@ define('ext.wikia.adEngine.adContext', [
 		return context.targeting.pageType === pageType;
 	}
 
+	function isPageFairDetectionEnabled() {
+		var isSupportedGeo = geo.isProperGeo(instantGlobals.wgAdDriverPageFairDetectionCountries);
+		return isUrlParamSet('pagefairdetection') || (isSupportedGeo && sampler.sample('pageFairDetection', 1, 10));
+	}
+
+	function isGSCEnabled() {
+		return abTest.getGroup('PROJECT_43') === 'GROUP_5' &&
+			geo.isProperGeo(instantGlobals.wgAdDriverGoogleConsumerSurveysCountries);
+	}
+
+	function isSourcePointDetectionDesktopEnabled(context) {
+		return context.opts.sourcePointDetectionUrl && (context.targeting.skin === 'oasis' &&
+			geo.isProperGeo(instantGlobals.wgAdDriverSourcePointDetectionCountries));
+	}
+
+	function isSourcePointDetectionMobile(context) {
+		return context.opts.sourcePointDetectionUrl && (context.targeting.skin === 'mercury' &&
+			geo.isProperGeo(instantGlobals.wgAdDriverSourcePointDetectionMobileCountries));
+	}
+
+	function updateRecoveryServicesAdContext(context, noExternals) {
```

Comment:
Also, what about `context.opts.showAds` ? We set it in case when Sony requests the page, couple of lines below. What about recovery in that case? My thinking is that we should not run recovery in that case. I'll ask Nef.